superset/db_engine_specs/bigquery.py

```diff
@@ -578,3 +578,17 @@ def _get_fields(cls, cols: List[Dict[str, Any]]) -> List[Any]:
         "author__name" and "author__email", respectively.
         """
         return [column(c["name"]).label(c["name"].replace(".", "__")) for c in cols]
+
+    @classmethod
+    def parse_error_exception(cls, exception: Exception) -> Exception:
+        try:
+            return Exception(
+                str(exception)  # pylint: disable=use-maxsplit-arg
+                .rsplit("\n")[0]
+                .rsplit(":")[1]
+                .strip()
+            )
+        except Exception:  # pylint: disable=broad-except
```

Comment:
I tried to get this to raise an exception and wasn't able to. I believe as long as the return value is a string, you should be able to perform all of the operations and don't need to put it in a try catch.

Reply:
Hmm if for some unknown reason the Exception has no `:` for example, accessing the [1] would cause a: `IndexError: list index out of range`.  Just adding the try/catch there in case anything like that happens with the message we get.